packages/dds/tree/src/tree/cursor.ts

```diff
@@ -218,6 +218,33 @@ export interface ITreeCursorSynchronous extends ITreeCursor {
     readonly pending: false;
 }
 
+/**
+ * @param cursor - tree whose fields will be visited.
+ * @param f - builds output from field, which will be selected in cursor when cursor is provided.
+ * If `f` moves cursor, it must put it back to where it was at the beginning of `f` before returning.
+ * @returns array resulting from applying `f` to each field of the current node on `cursor`.
+ * Returns an empty array if the node is empty or not present (which are considered the same).
+ * Note that order is not specified for field iteration.
+ */
+ export function mapCursorFields<T>(cursor: ITreeCursor, f: (cursor: ITreeCursor) => T): T[] {
```

Comment:
these kinds of functions should take in a generic cursor type, and provide it to the callback so they can be used with Forest specific cursors are needed.